addons/point_of_sale/static/src/js/Screens/ProductScreen/ProductsWidget.js

```diff
@@ -21,7 +21,12 @@ export class ProductsWidget extends Component {
      */
     setup() {
         super.setup();
-        this.state = useState({ previousSearchWord: "", currentOffset: 0 });
+        this.state = useState({
+            previousSearchWord: "",
+            currentOffset: 0,
+            showReloadMessage: false,
+            hasProducts: Object.keys(this.env.pos.db.product_by_id).length > 0
```

Comment:
Maybe we can make `hasProducts` a computed field but to keep it simple, it's okay this way. (Keeping this note for the future.)